src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1754 @@
+        # stage in which they were treated.
+        disability_series_for_alive_persons.loc[
+            (
+                ~pd.isnull(df.ce_date_treatment) & (
+                    (df.ce_hpv_cc_status == "stage1") |
+                    (df.ce_hpv_cc_status == "stage2a") |
+                    (df.ce_hpv_cc_status == "stage2b") |
+                    (df.ce_hpv_cc_status == "stage3")
+                ) & (df.ce_hpv_cc_status == df.ce_stage_at_which_treatment_given)
+            )
+        ] = self.daly_wts['stage_1_3_treated']
+
+        # Assign daly_wt to those in stage4 cancer (who have not had palliative care)
+        disability_series_for_alive_persons.loc[
+            (df.ce_hpv_cc_status == "stage4") &
+            (pd.isnull(df.ce_date_palliative_care))
+            ] = self.daly_wts['stage4']
+
+        # Assign daly_wt to those in stage4 cancer, who have had palliative care
+        disability_series_for_alive_persons.loc[
+            (df.ce_hpv_cc_status == "stage4") &
+            (~pd.isnull(df.ce_date_palliative_care))
+            ] = self.daly_wts['stage4_palliative_care']
+
+        return disability_series_for_alive_persons
+
+    def do_at_generic_first_appt(
+        self,
+        person_id: int,
+        individual_properties: IndividualProperties,
+        symptoms: List[str],
+        schedule_hsi_event: HSIEventScheduler,
+        **kwargs,
+    ) -> None:
+        if 'vaginal_bleeding' in symptoms:
+            schedule_hsi_event(
+                HSI_CervicalCancerPresentationVaginalBleeding(
+                    person_id=person_id,
+                    module=self
+                ),
+                priority=0,
+                topen=self.sim.date,
+                tclose=None)
+
+    def perform_cin_procedure(self, hsi_event, person_id):
+        """Function to decide treatment for individuals with CIN based on year. If year is >= transition_testing_year then Thermoablation, else  Cryotherapy
+        :param hsi_event: HSI Event (required to pass in order to register equipment)
+        :param person_id: person of interest
+        """
+        year = self.sim.date.year
+        p = self.parameters
+        hs = self.sim.modules["HealthSystem"]
+        treatment_methods = {
+            'Thermoablation': {
+                'event_class': HSI_CervicalCancer_Thermoablation_CIN
+            },
+            'Cryotherapy': {
+                'event_class': HSI_CervicalCancer_Cryotherapy_CIN
+            }
+        }
+
+        selected_method = 'Thermoablation' if year >= p['transition_testing_year'] else 'Cryotherapy'
+        method_info = treatment_methods[selected_method]
+
+        # todo Review addition of equipment Thermoablation Device, currently setting to LLETZ Machine
+        hsi_event.add_equipment({'Cusco’s/ bivalved Speculum (small, medium, large)'})
+        hsi_event.add_equipment({'LLETZ Machines'} if selected_method == 'Thermoablation' else {'Cryotherapy unit'})
```

Comment:
is is right the equipment attaches to this HSI and do the `HSI_CervicalCancer_Thermoablation_CIN` or `HSI_CervicalCancer_Cryotherapy_CIN`.....?